### PR TITLE
InitializableModules: Update CLI

### LINKED ISSUE #706: InitializableModules: Update CLI to raise a warning when interacting with non-initialized modules

### PR DESCRIPTION
**Pull request by leomassazza**
Fix #706

The PR adds the ability to check if the contract selected is initializable and if it is already initialized, raising a Warn message if not. That is accomplished with the introduction of a new subtask that is executed in the pick-function SUBTASK before showing the list of functions available.

### CODE REVIEW

File: packages/cli/subtasks/pick-function.js [hunk 1]
```diff
@@ -21,6 +21,8 @@ subtask(SUBTASK_PICK_FUNCTION, 'Pick a function from the given contract').setAct
       };
     });
 
+    await hre.run(SUBTASK_CHECK_INITIALIZATION);
```
Comment (leomassazza) [inline packages/cli/subtasks/pick-function.js:24]: Do the checking before showing the functions

**Review by mjlescano — APPROVED** [review]

File: packages/cli/subtasks/check-initializable.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  SUBTASK_CHECK_INITIALIZATION,
+  'Check if current contract is initializable and if it is initialized'
+).setAction(async (taskArguments, hre) => {
+  const target = hre.deployer.deployment.general.contracts[hre.cli.contractName];
+  const address = target.proxyAddress || target.deployedAddress;
+  const abi = hre.deployer.deployment.abis[hre.cli.contractName];
+  const abiFunctions = abi.filter((abiItem) => abiItem.name && abiItem.type === 'function');
+
+  const capitalizedContractName = capitalize(hre.cli.contractName.split(':')[1]);
+  const isInitializableFunctionName = `is${capitalizedContractName}Initialized`;
```
Comment (ajsantander) [inline packages/cli/subtasks/check-initializable.js:16]: Small semantic improvement: this should be called isInitializedFunctionName. Its existance determines if the contract is initializable. Becomes more relevant when you actually call the function a few lines below.

File: packages/cli/subtasks/check-initializable.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  const initializeFunctionName = `initialize${capitalizedContractName}`;
+
+  const isInitializable = abiFunctions.some((f) => f.name === isInitializableFunctionName);
+
+  if (isInitializable) {
+    const contract = new hre.ethers.Contract(address, abi, hre.ethers.provider);
+    const result = await contract[isInitializableFunctionName]();
```
Comment (ajsantander) [inline packages/cli/subtasks/check-initializable.js:23]: Here’s where the small semantic problem pointed out above makes the code weird to read.

**Review by ajsantander — APPROVED** [review]
Awesome! Just a mini nit

File: packages/cli/subtasks/check-initializable.js (hunk 2 above)
Comment (leomassazza) [reply packages/cli/subtasks/check-initializable.js:16]: in fact, is not semantic, it was the wrong name (a mental typo) :)

File: packages/cli/subtasks/check-initializable.js (hunk 3 above)
Comment (leomassazza) [reply packages/cli/subtasks/check-initializable.js]: fixed

**Pull request merged**